hack/create-dev-cluster.sh

```diff
@@ -62,14 +62,25 @@ capz::util::generate_ssh_key
 echo "================ DOCKER BUILD ==============="
 PULL_POLICY=IfNotPresent make modules docker-build
 
-echo "================ MAKE CLEAN ==============="
-make clean
+setup() {
+    echo "================ MAKE CLEAN ==============="
+    make clean
 
-echo "================ KIND RESET ==============="
-make kind-reset
+    echo "================ KIND RESET ==============="
+    make kind-reset
 
-echo "================ INSTALL TOOLS ==============="
-make install-tools
+    echo "================ INSTALL TOOLS ==============="
+    make install-tools
+}
 
-echo "================ CREATE CLUSTER ==============="
-make create-cluster
+create_cluster() {
+    echo "================ CREATE CLUSTER ==============="
+    make create-cluster
+}
+
+setup
+
+until create_cluster; do
```

Comment:
I don't think a retry will actually work if we don't also delete the management cluster, because it will try to re-create the management cluster (we could test this?)